pkg/generate/code/set_sdk.go

```diff
@@ -190,6 +190,22 @@ func SetSDK(
 			}
 		}
 
+		fc, ok := fieldConfigs[memberName]
+		if ok && fc.IsSecret {
+			out += fmt.Sprintf("%sif %s.Spec.%s != nil {\n", indent, sourceVarName, memberName)
+			out += fmt.Sprintf("%s%sresp, err := rm.rr.SecretValueFromReference(ctx, %s.Spec.%s)\n", indent,
+				indent, sourceVarName, memberName)
+			out += fmt.Sprintf("%s%sif err != nil {\n", indent, indent)
+			out += fmt.Sprintf("%s%s%sreturn nil, err\n", indent, indent, indent)
+			out += fmt.Sprintf("%s%s}\n", indent, indent)
+			out += fmt.Sprintf("%s%sif resp != nil {\n", indent, indent)
+			out += fmt.Sprintf("%s%s%s%s.Set%s(%s)\n", indent, indent, indent,
```

Comment:
see my comment on the runtime PR, but this will need to change to pass the returned secret value as-is instead of dereferencing a string pointer.